### PR TITLE
Strong name assemblies (Fix for #15)

### PR DESCRIPTION
**Pull request by anthonylangsworth**
Merge or rebase this into the 0.6.4 (or latest) branch to strong name the main and unit test assemblies.

Fix for #15.

### CODE REVIEW

File: src/VaultSharp/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -34,5 +34,4 @@
 // [assembly: AssemblyVersion("1.0.*")]
 [assembly: AssemblyVersion("0.6.1.0")]
 [assembly: AssemblyFileVersion("0.6.1.0")]
-[assembly: InternalsVisibleTo("VaultSharp.UnitTests")]
-[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]
```
Comment (rajanadar) [inline src/VaultSharp/Properties/AssemblyInfo.cs]: did the moq based tests run without this attribute? this attribute is needed for the moqing of internal classes.

File: src/VaultSharp/VaultSharp.csproj [hunk 2]
```diff
@@ -35,6 +35,12 @@
     <WarningLevel>4</WarningLevel>
     <DocumentationFile>bin\Release\VaultSharp.XML</DocumentationFile>
   </PropertyGroup>
+  <PropertyGroup>
+    <SignAssembly>true</SignAssembly>
+  </PropertyGroup>
+  <PropertyGroup>
+    <AssemblyOriginatorKeyFile>VaultSharp.snk</AssemblyOriginatorKeyFile>
```
Comment (rajanadar) [inline src/VaultSharp/VaultSharp.csproj]: can you please create a new folder called 'keys' at the level of src and test and put this dummy VaultSharp.snk file there? and then link it from both the projects.

Comment (rajanadar) [inline src/VaultSharp/VaultSharp.csproj]: meanwhile, i'll procure a new pfx and replace the snk with a real one, before i publish 0.6.4 or the nuget.

**Comment by rajanadar** [comment]
thanks @anthonylangsworth for doing this.

File: src/VaultSharp/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (anthonylangsworth) [inline src/VaultSharp/Properties/AssemblyInfo.cs]: Fixed in 2296b55912f29d6978ca4463bab9eb191a78ab49.

File: src/VaultSharp/VaultSharp.csproj (hunk 2 above)
Comment (anthonylangsworth) [inline src/VaultSharp/VaultSharp.csproj]: Moved the snk file to the keys folder in fe249a9ffd72befbcb6c7b480eb9fbca30c39725.

While you are welcome to replace the snk file, there is no point. The requirement, at least from my team, is the VaultSharp assemblies downloaded via NuGet are strong named. The actual snk file is irrelevant. As discussed previously, customers using VaultSharp can compile it from source, strong naming it with their own snk file, if required.

Similarly, digitally signing the assembly (e.g. using `signtool.exe` with a pfx file) is not required. While I have not specifically looked, I am not aware of any open source projects that do so, particularly when you need to purchase a certificate and private key. Once again. customers can procure their own certificate and private key if a digital signature is required.

Comment (anthonylangsworth) [inline src/VaultSharp/VaultSharp.csproj]: ... and just to be clear, the snk file I added was created using `sn -k VaultSharp.snk`.

Comment (rajanadar) [inline src/VaultSharp/VaultSharp.csproj]: cool. i agree that digitally signing the assembly is not required.

but to generate the strong named assembly for NuGet I still need to procure/purchase a cert.

Comment (rajanadar) [inline src/VaultSharp/VaultSharp.csproj]: shouldn't the path to this key file change to the keys folder?

Comment (rajanadar) [inline src/VaultSharp/VaultSharp.csproj]: maybe i can hold off on a strong named nuget package. 
i am assuming this Pull Request, when merged, will solve your issue. 

i can evaluate others' needs on a case by case basis. the world seems to be divided when it comes to strong naming open source projects in NuGet.

Comment (anthonylangsworth) [inline src/VaultSharp/VaultSharp.csproj]: I thought the reference would be the path, too, but it works as is.

Comment (anthonylangsworth) [inline src/VaultSharp/VaultSharp.csproj]: You do not need to procure a certificate to strong name an assembly, whether it is distributed via NuGet or other means. Perhaps you are getting confused with creating a digital signature using `signtool.exe`.

Merely merging this pull request will not solve my team's issue. We are referencing VaultSharp from a strong named assembly, meaning VaultSharp must be strong named. That is why I also strong named the VaultSharp test assembly. We would prefer to manage references via NuGet to take advantage of automatic upgrades. Many open source projects distribute strong named versions of assemblies, as mentioned above.

Please let me know if anything is unclear.

**Comment by rajanadar** [comment]
@anthonylangsworth this change is now in.

i spent some time, to think about the NuGet package release strategy. 

- whether to hold off till 1.0.0 - major version change, OR 
- publish 2 packages - SN and non-SN OR
- just break the NuGet compatibility and publish 0.6.4 as a SN assembly only.

have decided to go with the last option. one nuget package, SN only.
this is going to be painful for the existing few thousand (low) consumers, but probably better to absorb now than later.

**Comment by rajanadar** [comment]
i uncovered a couple of other issues, as part of my 0.6.4 acceptance testing.
coupled with testing out my SN package's behavior (using local nuget), i am pushing the release of 0.6.4 by a week for some more QA.

**Comment by rajanadar** [comment]
@anthonylangsworth VaultSharp 0.6.4 is now live. please go get it to see this fix!

**Comment by anthonylangsworth** [comment]
I have successfully referenced VaultSharp 0.6.4 from a strong named test assembly. Looks like it works. Thanks!

**Pull request merged**